### PR TITLE
Allow scala libraries as deps of scalapb_proto_library

### PR DESCRIPTION
**Pull request by azymnis**
Addresses issue #316

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

File: scala_proto/scala_proto.bzl [hunk 1]
```diff
@@ -316,21 +316,18 @@ def _colon_paths(data):
...
+    proto_deps, jvm_deps = [], []
     for target in ctx.attr.deps:
         if hasattr(target, 'proto'):
             proto_deps.append(target)
             acc_imports += target.proto.transitive_sources
         else:
-            java_proto_lib_deps.append(target)
+            jvm_deps.append(target)
 
-    if not ctx.attr.with_java and len(java_proto_lib_deps) > 0:
```
Comment (ittaiz) [inline scala_proto/scala_proto.bzl]: this removal is related to #315 ?

Comment (azymnis) [reply scala_proto/scala_proto.bzl]: No I hadn't seen that issue. I'll need to investigate that too

Comment (ittaiz) [reply scala_proto/scala_proto.bzl]: so why did you remove it? not saying it's wrong I just don't understand how it's related to the current change

Comment (azymnis) [reply scala_proto/scala_proto.bzl]: Because I want to allow for scala_library deps even if with_java is false

Comment (ittaiz) [reply scala_proto/scala_proto.bzl]: oh I see. thanks

**Comment by ittaiz** [comment]
LGTM, @azymnis any objections to merge?

**Comment by azymnis** [comment]
@ittaiz please merge. I'll keep investigating ongoing issues with these and add fixes in a separate branch

**Comment by BlueDragonX** [comment]
Waiting for this to be merged so I can issue a PR for [this](https://github.com/zenreach/rules_scala/commit/ce7972f0d2046f5be02e23e8e21b8b4827bbb485).

**Pull request merged**